Bug: Error executing this data block https://registry.terraform.io/providers/mulesoft-anypoint/anypoint/latest/docs/data-sources/amq

`Stack trace from the terraform-provider-anypoint_v1.8.4 plugin:
panic: interface conversion: interface {} is int, not int32
goroutine 67 [running]:
[github.com/mulesoft-anypoint/terraform-provider-anypoint/anypoint.parseAMQSearchOpts](http://github.com/mulesoft-anypoint/terraform-provider-anypoint/anypoint.parseAMQSearchOpts)({{0x1261938, 0x429843d8e40}, 0x429842579d0, {0x429843102d0, 0x24}, {0x429843102a0, 0x24}, {0x4298447b350, 0xc}, 0x429843de7f0, ...}, ...)
[github.com/mulesoft-anypoint/terraform-provider-anypoint/anypoint/data_source_amq.go:208](http://github.com/mulesoft-anypoint/terraform-provider-anypoint/anypoint/data_source_amq.go:208) +0x974
[github.com/mulesoft-anypoint/terraform-provider-anypoint/anypoint.dataSourceAMQRead](http://github.com/mulesoft-anypoint/terraform-provider-anypoint/anypoint.dataSourceAMQRead)({0x12619e0, 0x42984440d90}, 0x429844cb380, {0x11d8a60?, 0x42984358240?})
[github.com/mulesoft-anypoint/terraform-provider-anypoint/anypoint/data_source_amq.go:152](http://github.com/mulesoft-anypoint/terraform-provider-anypoint/anypoint/data_source_amq.go:152) +0x413
[github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema](http://github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema).(*Resource).read(0x429840b81c0, {0x1261938, 0x429843295c0}, 0x429844cb380, {0x11d8a60, 0x42984358240})
[github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/resource.go:724](http://github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/resource.go:724) +0x119
[github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema](http://github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema).(*Resource).ReadDataApply(0x429840b81c0, {0x1261938, 0x429843295c0}, 0x429844cb280, {0x11d8a60, 0x42984358240})
[github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/resource.go:943](http://github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/resource.go:943) +0x127
[github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema](http://github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema).(*GRPCProviderServer).ReadDataSource(0x42983f67d40, {0x1261938?, 0x42984329500?}, 0x429846035c0)
[github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/grpc_provider.go:1179](http://github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/grpc_provider.go:1179) +0x349
[github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server](http://github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server).(*server).ReadDataSource(0x42984134460, {0x1261938?, 0x42984328d20?}, 0x42983edb900)
[github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov5/tf5server/server.go:658](http://github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov5/tf5server/server.go:658) +0x3ca
[github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadDataSource_Handler](http://github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadDataSource_Handler)({0x1196e20, 0x42984134460}, {0x1261938, 0x42984328d20}, 0x4298414c850, 0x0)
[github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:421](http://github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:421) +0x1a6
[google.golang.org/grpc](http://google.golang.org/grpc).(*Server).processUnaryRPC(0x42983f9e1e0, {0x126a7e0, 0x42983ebc000}, 0x42984323680, 0x42983fc33b0, 0x1c48f30, 0x0)
[google.golang.org/grpc@v1.48.0/server.go:1295](http://google.golang.org/grpc@v1.48.0/server.go:1295) +0xaa9
[google.golang.org/grpc](http://google.golang.org/grpc).(*Server).handleStream(0x42983f9e1e0, {0x126a7e0, 0x42983ebc000}, 0x42984323680, 0x0)
[google.golang.org/grpc@v1.48.0/server.go:1636](http://google.golang.org/grpc@v1.48.0/server.go:1636) +0x9c8
[google.golang.org/grpc](http://google.golang.org/grpc).(*Server).serveStreams.func1.2()
[google.golang.org/grpc@v1.48.0/server.go:932](http://google.golang.org/grpc@v1.48.0/server.go:932) +0x87
created by [google.golang.org/grpc](http://google.golang.org/grpc).(*Server).serveStreams.func1 in goroutine 22
[google.golang.org/grpc@v1.48.0/server.go:930](http://google.golang.org/grpc@v1.48.0/server.go:930) +0x226
Error: The terraform-provider-anypoint_v1.8.4 plugin crashed!
This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
::error::Terraform exited with code 1.`

--- a/anypoint/data_source_ame.go
+++ b/anypoint/data_source_ame.go
@@ -177,19 +177,19 @@ func parseAMESearchOpts(req amq.DefaultApiApiGetAMQListRequest, params *schema.S
 
 	for k, v := range opts.(map[string]any) {
 		if k == "offset" {
-			req = req.Offset(int32(v.(int)))
+			req = req.Offset(toInt32(v))
 			continue
 		}
 		if k == "limit" {
-			req = req.Limit(int32(v.(int)))
+			req = req.Limit(toInt32(v))
 			continue
 		}
-		if k == "starts_with" && v.(string) != "" {
-			req = req.StartsWith(v.(string))
+		if k == "starts_with" && toString(v) != "" {
+			req = req.StartsWith(toString(v))
 			continue
 		}
 		if k == "destination_ids" {
-			req = req.DestinationIds(ListInterface2ListStrings(v.([]any)))
+			req = req.DestinationIds(toStringSlice(v))
 			continue
 		}
 	}

--- a/anypoint/data_source_amq.go
+++ b/anypoint/data_source_amq.go
@@ -205,19 +205,19 @@ func parseAMQSearchOpts(req amq.DefaultApiApiGetAMQListRequest, params *schema.S
 
 	for k, v := range opts.(map[string]any) {
 		if k == "offset" {
-			req = req.Offset(int32(v.(int)))
+			req = req.Offset(toInt32(v))
 			continue
 		}
 		if k == "limit" {
-			req = req.Limit(int32(v.(int)))
+			req = req.Limit(toInt32(v))
 			continue
 		}
-		if k == "starts_with" && v.(string) != "" {
-			req = req.StartsWith(v.(string))
+		if k == "starts_with" && toString(v) != "" {
+			req = req.StartsWith(toString(v))
 			continue
 		}
 		if k == "destination_ids" {
-			req = req.DestinationIds(ListInterface2ListStrings(v.([]any)))
+			req = req.DestinationIds(toStringSlice(v))
 			continue
 		}
 	}

--- a/anypoint/data_source_apim.go
+++ b/anypoint/data_source_apim.go
@@ -499,19 +499,19 @@ func parseApimSearchOpts(req apim.DefaultApiGetEnvApimInstancesRequest, params *
 			continue
 		}
 		if k == "offset" {
-			req = req.Offset(int32(v.(int)))
+			req = req.Offset(toInt32(v))
 			continue
 		}
 		if k == "limit" {
-			req = req.Limit(int32(v.(int)))
+			req = req.Limit(toInt32(v))
 			continue
 		}
-		if k == "sort" && v.(string) != "" {
-			req = req.Sort(v.(string))
+		if k == "sort" && toString(v) != "" {
+			req = req.Sort(toString(v))
 			continue
 		}
 		if k == "ascending" {
-			req = req.Ascending(v.(bool))
+			req = req.Ascending(toBool(v))
 			continue
 		}
 	}

--- a/anypoint/data_source_app_deployments_v2.go
+++ b/anypoint/data_source_app_deployments_v2.go
@@ -211,16 +211,16 @@ func parseAppDeploymentSearchOpts(req application_manager_v2.DefaultApiGetAllDep
 	}
 	opts := params.List()[0]
 	for k, v := range opts.(map[string]any) {
-		if k == "target_id" {
-			req = req.TargetId(v.(string))
+		if k == "target_id" && toString(v) != "" {
+			req = req.TargetId(toString(v))
 			continue
 		}
 		if k == "offset" {
-			req = req.Offset(int32(v.(int)))
+			req = req.Offset(toInt32(v))
 			continue
 		}
 		if k == "limit" {
-			req = req.Limit(int32(v.(int)))
+			req = req.Limit(toInt32(v))
 			continue
 		}
 	}

--- a/anypoint/data_source_connected_apps.go
+++ b/anypoint/data_source_connected_apps.go
@@ -245,16 +245,16 @@ func parseConnectedAppSearchOpts(req connected_app.DefaultApiGetAllConnectedApps
 	}
 	opts := params.List()[0]
 	for k, v := range opts.(map[string]any) {
-		if k == "search" && v.(string) != "" {
-			req = req.Search(v.(string))
+		if k == "search" && toString(v) != "" {
+			req = req.Search(toString(v))
 			continue
 		}
 		if k == "offset" {
-			req = req.Offset(int32(v.(int)))
+			req = req.Offset(toInt32(v))
 			continue
 		}
 		if k == "limit" {
-			req = req.Limit(int32(v.(int)))
+			req = req.Limit(toInt32(v))
 			continue
 		}
 	}

--- a/anypoint/data_source_exchange_policy_templates.go
+++ b/anypoint/data_source_exchange_policy_templates.go
@@ -287,28 +287,28 @@ func parseExchangePolicyTemplatesSearchOpts(req apim_policy.DefaultApiGetOrgExch
 	}
 	opts := params.List()[0]
 	for k, v := range opts.(map[string]any) {
-		if k == "env_id" {
-			req = req.EnvironmentId(v.(string))
+		if k == "env_id" && toString(v) != "" {
+			req = req.EnvironmentId(toString(v))
 			continue
 		}
 		if k == "split_model" {
-			req = req.SplitModel(v.(bool))
+			req = req.SplitModel(toBool(v))
 			continue
 		}
 		if k == "latest" {
-			req = req.Latest(v.(bool))
+			req = req.Latest(toBool(v))
 			continue
 		}
-		if k == "api_instance_id" {
-			req = req.ApiInstanceId(v.(string))
+		if k == "api_instance_id" && toString(v) != "" {
+			req = req.ApiInstanceId(toString(v))
 			continue
 		}
 		if k == "include_configuration" {
-			req = req.IncludeConfiguration(v.(bool))
+			req = req.IncludeConfiguration(toBool(v))
 			continue
 		}
 		if k == "automated_only" {
-			req = req.AutomatedOnly(v.(bool))
+			req = req.AutomatedOnly(toBool(v))
 			continue
 		}
 	}

--- a/anypoint/data_source_roles.go
+++ b/anypoint/data_source_roles.go
@@ -207,32 +207,32 @@ func parseRoleSearchOpts(req role.DefaultApiApiRolesGetRequest, params *schema.S
 	opts := params.List()[0]
 
 	for k, v := range opts.(map[string]any) {
-		if k == "name" && len(v.(string)) > 0 {
-			req = req.Name(v.(string))
+		if k == "name" && len(toString(v)) > 0 {
+			req = req.Name(toString(v))
 			continue
 		}
-		if k == "description" && len(v.(string)) > 0 {
-			req = req.Description(v.(string))
+		if k == "description" && len(toString(v)) > 0 {
+			req = req.Description(toString(v))
 			continue
 		}
 		if k == "include_internal" {
-			req = req.IncludeInternal(v.(bool))
+			req = req.IncludeInternal(toBool(v))
 			continue
 		}
-		if k == "search" && len(v.(string)) > 0 {
-			req = req.Search(v.(string))
+		if k == "search" && len(toString(v)) > 0 {
+			req = req.Search(toString(v))
 			continue
 		}
 		if k == "offset" {
-			req = req.Offset(int32(v.(int)))
+			req = req.Offset(toInt32(v))
 			continue
 		}
 		if k == "limit" {
-			req = req.Limit(int32(v.(int)))
+			req = req.Limit(toInt32(v))
 			continue
 		}
 		if k == "ascending" {
-			req = req.Ascending(v.(bool))
+			req = req.Ascending(toBool(v))
 			continue
 		}
 	}

--- a/anypoint/data_source_secretgroup_keystores.go
+++ b/anypoint/data_source_secretgroup_keystores.go
@@ -154,8 +154,8 @@ func parseSgKeystoreSearchOpts(req secretgroup_keystore.DefaultApiGetSecretGroup
 	}
 	opts := params.List()[0]
 	for k, v := range opts.(map[string]any) {
-		if k == "type" {
-			req = req.Type_(v.(string))
+		if k == "type" && toString(v) != "" {
+			req = req.Type_(toString(v))
 			continue
 		}
 	}

--- a/anypoint/data_source_secretgroup_truststores.go
+++ b/anypoint/data_source_secretgroup_truststores.go
@@ -193,8 +193,8 @@ func parseSgTruststoreSearchOpts(req secretgroup_truststore.DefaultApiGetSecretG
 	}
 	opts := params.List()[0]
 	for k, v := range opts.(map[string]any) {
-		if k == "type" {
-			req = req.Type_(v.(string))
+		if k == "type" && toString(v) != "" {
+			req = req.Type_(toString(v))
 			continue
 		}
 	}

--- a/anypoint/data_source_secretgroups.go
+++ b/anypoint/data_source_secretgroups.go
@@ -227,7 +227,7 @@ func parseSecretGroupsSearchOpts(req secretgroup.DefaultApiGetEnvSecretGroupsReq
 	opts := params.List()[0]
 	for k, v := range opts.(map[string]any) {
 		if k == "downloadable" {
-			req = req.Downloadable(v.(bool))
+			req = req.Downloadable(toBool(v))
 			continue
 		}
 	}

--- a/anypoint/data_source_team_groupmappings.go
+++ b/anypoint/data_source_team_groupmappings.go
@@ -155,11 +155,11 @@ func parseTeamGroupMappingsSearchOpts(req team_group_mappings.DefaultApiApiOrgan
 
 	for k, v := range opts.(map[string]any) {
 		if k == "offset" {
-			req = req.Offset(int32(v.(int)))
+			req = req.Offset(toInt32(v))
 			continue
 		}
 		if k == "limit" {
-			req = req.Limit(int32(v.(int)))
+			req = req.Limit(toInt32(v))
 			continue
 		}
 	}

--- a/anypoint/data_source_team_members.go
+++ b/anypoint/data_source_team_members.go
@@ -206,36 +206,39 @@ func parseTeamMembersSearchOpts(req team_members.DefaultApiApiOrganizationsOrgId
 	opts := params.List()[0]
 
 	for k, v := range opts.(map[string]any) {
-		if k == "membership_type" && v.(string) != "" {
-			req = req.MembershipType(v.(string))
+		if k == "membership_type" && toString(v) != "" {
+			req = req.MembershipType(toString(v))
 			continue
 		}
-		if k == "identity_type" && v.(string) != "" {
-			req = req.IdentityType(v.(string))
+		if k == "identity_type" && toString(v) != "" {
+			req = req.IdentityType(toString(v))
 			continue
 		}
-		if k == "member_ids" && len(v.([]any)) > 0 {
-			req = req.MemberIds(ListInterface2ListStrings(v.([]any)))
+		if k == "member_ids" {
+			ids := toStringSlice(v)
+			if len(ids) > 0 {
+				req = req.MemberIds(ids)
+			}
 			continue
 		}
-		if k == "search" && v.(string) != "" {
-			req = req.Search(v.(string))
+		if k == "search" && toString(v) != "" {
+			req = req.Search(toString(v))
 			continue
 		}
 		if k == "offset" {
-			req = req.Offset(int32(v.(int)))
+			req = req.Offset(toInt32(v))
 			continue
 		}
 		if k == "limit" {
-			req = req.Limit(int32(v.(int)))
+			req = req.Limit(toInt32(v))
 			continue
 		}
-		if k == "sort" && v.(string) != "" {
-			req = req.Sort(v.(string))
+		if k == "sort" && toString(v) != "" {
+			req = req.Sort(toString(v))
 			continue
 		}
 		if k == "ascending" {
-			req = req.Ascending(v.(bool))
+			req = req.Ascending(toBool(v))
 			continue
 		}
 	}

--- a/anypoint/data_source_team_roles.go
+++ b/anypoint/data_source_team_roles.go
@@ -178,20 +178,20 @@ func parseTeamRolesSearchOpts(req team_roles.DefaultAPIGetTeamRolesRequest, para
 	opts := params.List()[0]
 
 	for k, v := range opts.(map[string]any) {
-		if k == "role_id" && v.(string) != "" {
-			req = req.RoleId(v.(string))
+		if k == "role_id" && toString(v) != "" {
+			req = req.RoleId(toString(v))
 			continue
 		}
-		if k == "search" && v.(string) != "" {
-			req = req.Search(v.(string))
+		if k == "search" && toString(v) != "" {
+			req = req.Search(toString(v))
 			continue
 		}
 		if k == "offset" {
-			req = req.Offset(int32(v.(int)))
+			req = req.Offset(toInt32(v))
 			continue
 		}
 		if k == "limit" {
-			req = req.Limit(int32(v.(int)))
+			req = req.Limit(toInt32(v))
 			continue
 		}
 	}

--- a/anypoint/data_source_teams.go
+++ b/anypoint/data_source_teams.go
@@ -238,32 +238,32 @@ func parseTeamSearchOpts(req team.DefaultAPIGetTeamsRequest, params *schema.Set)
 			req = req.ParentTeamId(ListInterface2ListStrings(v.([]any)))
 			continue
 		}
-		if k == "team_id" && v.(string) != "" {
-			req = req.TeamId(v.(string))
+		if k == "team_id" && toString(v) != "" {
+			req = req.TeamId(toString(v))
 			continue
 		}
-		if k == "team_type" && v.(string) != "" {
-			req = req.TeamType(v.(string))
+		if k == "team_type" && toString(v) != "" {
+			req = req.TeamType(toString(v))
 			continue
 		}
-		if k == "search" && v.(string) != "" {
-			req = req.Search(v.(string))
+		if k == "search" && toString(v) != "" {
+			req = req.Search(toString(v))
 			continue
 		}
 		if k == "offset" {
-			req = req.Offset(int32(v.(int)))
+			req = req.Offset(toInt32(v))
 			continue
 		}
 		if k == "limit" {
-			req = req.Limit(int32(v.(int)))
+			req = req.Limit(toInt32(v))
 			continue
 		}
-		if k == "sort" && v.(string) != "" {
-			req = req.Sort(v.(string))
+		if k == "sort" && toString(v) != "" {
+			req = req.Sort(toString(v))
 			continue
 		}
 		if k == "ascending" {
-			req = req.Ascending(v.(bool))
+			req = req.Ascending(toBool(v))
 			continue
 		}
 	}

--- a/anypoint/data_source_user_rolegroups.go
+++ b/anypoint/data_source_user_rolegroups.go
@@ -210,11 +210,11 @@ func parseUserRolegroupsSearchOpts(req user_rolegroups.DefaultApiApiOrganization
 
 	for k, v := range opts.(map[string]any) {
 		if k == "offset" {
-			req = req.Offset(int32(v.(int)))
+			req = req.Offset(toInt32(v))
 			continue
 		}
 		if k == "limit" {
-			req = req.Limit(int32(v.(int)))
+			req = req.Limit(toInt32(v))
 			continue
 		}
 	}

--- a/anypoint/data_source_users.go
+++ b/anypoint/data_source_users.go
@@ -254,15 +254,15 @@ func parseUsersSearchOpts(req user.DefaultApiApiOrganizationsOrgIdUsersGetReques
 
 	for k, v := range opts.(map[string]any) {
 		if k == "offset" {
-			req = req.Offset(int32(v.(int)))
+			req = req.Offset(toInt32(v))
 			continue
 		}
 		if k == "limit" {
-			req = req.Limit(int32(v.(int)))
+			req = req.Limit(toInt32(v))
 			continue
 		}
-		if k == "type" {
-			req = req.Type_(v.(string))
+		if k == "type" && toString(v) != "" {
+			req = req.Type_(toString(v))
 			continue
 		}
 	}

--- a/anypoint/util.go
+++ b/anypoint/util.go
@@ -67,6 +67,78 @@ func ListInterface2ListStrings(array []any) []string {
 	return list
 }
 
+// toInt32 safely converts any numeric value to int32 without panicking.
+// Handles int, int32, int64, float32, float64, and string representations.
+func toInt32(v any) int32 {
+	if v == nil {
+		return 0
+	}
+	switch i := v.(type) {
+	case int:
+		return int32(i)
+	case int8:
+		return int32(i)
+	case int16:
+		return int32(i)
+	case int32:
+		return i
+	case int64:
+		return int32(i)
+	case uint:
+		return int32(i)
+	case uint8:
+		return int32(i)
+	case uint16:
+		return int32(i)
+	case uint32:
+		return int32(i)
+	case uint64:
+		return int32(i)
+	case float32:
+		return int32(i)
+	case float64:
+		return int32(i)
+	case string:
+		val, _ := strconv.Atoi(i)
+		return int32(val)
+	default:
+		return 0
+	}
+}
+
+// toString safely converts any value to string without panicking.
+func toString(v any) string {
+	if v == nil {
+		return ""
+	}
+	if s, ok := v.(string); ok {
+		return s
+	}
+	return fmt.Sprint(v)
+}
+
+// toBool safely converts any value to bool without panicking.
+func toBool(v any) bool {
+	if v == nil {
+		return false
+	}
+	if b, ok := v.(bool); ok {
+		return b
+	}
+	return false
+}
+
+// toStringSlice safely converts any value to []string without panicking.
+func toStringSlice(v any) []string {
+	if v == nil {
+		return nil
+	}
+	if arr, ok := v.([]any); ok {
+		return ListInterface2ListStrings(arr)
+	}
+	return nil
+}
+
 // tests if the provided value matches the value of an element in the valid slice. Will test with strings.EqualFold if ignoreCase is true
 func StringInSlice(expected []string, v string, ignoreCase bool) bool {
 	for _, e := range expected {

